srv/api/chat/edit.ts

```diff
@@ -26,13 +28,27 @@ export const updateChat = handle(async ({ params, body, user }) => {
   const prev = await store.chats.getChatOnly(id)
   if (prev?.userId !== user?.userId) throw errors.Forbidden
 
-  const chat = await store.chats.update(id, {
-    ...body,
-    greeting: body.greeting ?? '',
-    scenario: body.scenario ?? '',
-    sampleChat: body.sampleChat ?? '',
-    overrides: body.overrides ?? '',
-  })
+  const update: PartialUpdate<AppSchema.Chat> = {
+    name: body.name,
+    mode: body.mode,
+    adapter: body.adapter,
+  }
+
+  if (body.useOverrides === false) {
+    update.overrides = null
+    update.greeting = null
+    update.scenario = null
+    update.sampleChat = null
+  }
+
+  if (body.useOverrides === true) {
+    update.greeting = body.greeting || ''
+    update.scenario = body.scenario || ''
+    update.sampleChat = body.sampleChat || ''
+    update.overrides = body.overrides
```

Comment:
what if the user's override is to completely remove the persona field down to an empty string? the app will think greeting/scenario/samplechat overrides are disabled.